src/main/java/core/basesyntax/utils/generate/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.utils.generate.impl;
+
+import core.basesyntax.storage.impl.StorageImpl;
+import core.basesyntax.utils.generate.ReportGenerator;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private final ReportGenerator generator;
+
+    public ReportGeneratorImpl(ReportGenerator generator) {
+        this.generator = generator;
+    }
```

Comment:
The constructor of `ReportGeneratorImpl` should not take a `ReportGenerator` as a parameter. If you are trying to apply the Decorator pattern, make sure to implement it correctly, or if it's not required, remove the unnecessary delegation.